pkg/segment/search/conditioncheck.go

```diff
@@ -141,24 +128,17 @@ func ApplyColumnarSearchQuery(query *SearchQuery, multiColReader *segread.MultiC
 			return false, finalErr
 		}
 	case MatchDictArraySingleColumn:
-		rawColVal, err := multiColReader.ReadRawRecordFromColumnFile(query.QueryInfo.ColName, blockNum, recordNum, qid)
+		rawColVal, err := multiColReader.ReadRawRecordFromColumnFile(queryInfoColKeyIndex, blockNum, recordNum, qid, false)
 		if err != nil {
 			return false, err
 		}
 		return writer.ApplySearchToDictArrayFilter(query.QueryInfo.KValDte, query.QueryInfo.QValDte, rawColVal, Equals, true, holderDte)
 	case MatchDictArrayAllColumns:
 		var atleastOneNonError bool
 		var finalErr error
-		for _, colInfo := range multiColReader.AllColums {
-
-			// we skip rawsearching for columns that are dict encoded,
-			// since we already search for them in the prior call to applyColumnarSearchUsingDictEnc
-			_, ok := dictEncColNames[colInfo.ColumnName]
-			if ok {
-				continue
-			}
+		for colKeyIndex := range allColKeyIndices {
 
-			rawColVal, err := multiColReader.ReadRawRecordFromColumnFile(colInfo.ColumnName, blockNum, recordNum, qid)
+			rawColVal, err := multiColReader.ReadRawRecordFromColumnFile(colKeyIndex, blockNum, recordNum, qid, false)
```

Comment:
Why are we searching `allColkeyIndices` when we are supposed to skip raw searching for `dictEncColNames`.

In `filtersearch.go` where we are building the `allColKeyIndices`, we used `multiColReader.AllColums` which will contain all the columns.  Check the code [here](https://github.com/siglens/siglens/pull/1428/files#diff-1c06c829ef6cb4902da6c66dee810690b4f50a3cd971683a29bba74f52cdff6eR200-R210).

But we just need to loop through over `cmiPassedNonDictColKeyIndices`, since we are supposed to skip `dictEncColNames`?  

If my reasoning is true we can avoid creating the `allColKeyIndices` altogether, I think.

Reply:
previously for `MatchDictArrayAllColumns` queryType we were doing `for _, colInfo := range multiColReader.AllColums {`

so I kept the same case here. For dictArray searching we do not skip dictEncColNames,

Reply:
yes, we are looping through `multiColReader.AllColums` but inside that we are checking if the column exists in `dictEncColNames` and if `true` we are skipping the raw search. See the below screenshot for `MatchDictArrayAllColumns`. See the below screenshot or check the respective code here: https://github.com/siglens/siglens/blob/develop/pkg/segment/search/conditioncheck.go#L152-L160


![image](https://github.com/user-attachments/assets/7165291b-4e29-42a1-91b5-3f1b53aeab7c)

Reply:
yup, you are right, the previous PR removed that dict_enc check and moved it one stack higher but that PR did not adjust this part of the code. Will fix it